### DISCUSSION TITLE
Using Control Layers

### DISCUSSION THREAD

**Discussion by Chepko932** [Acly/krita-ai-diffusion#527] [Q&A]
I want to use the depth control layer for hands, but cant really figure out how to use it correctly?
Also cant find any documentation about it.

**Comment by DrewAlderfer** [comment]
What have you tried? There's a fair amount of information out there about depth map inputs for controlnet.

If you could include information about how you are trying to use it that would make it a lot easier to help.

Maybe try combining a depth map with a lineart layer and see how messing with the control layer weights affects your output?

**Reply by Chepko932** [reply]
Yeah i know how to use them in Controlnet, but not in Krita.

I have an image with a "broken" hand, these are my steps
-Loading the image in Krita
-Selecting the Hand with the selection tool, create a new layer
-Click on "Add Control Layer", select "Hands" + the Background layer and hit the "Generate Control layer from Current image" (Which creates a depth map of the hand)
-Start the live preview, write "left hand" for example

-Outpot totally different from the control layer

**Reply by ETCore7** [reply]
Not sure what you are trying to do. When you load an image and create a depth map you get shades of gray that imply distances. An object in the back is dark. When you brighten the spot the engine will try to render the object or a similar shape more to the front.

(snipped partly incorrect info)

And also, whatever you do, try different strenghts. Some controlnets seem to do nothing with the default 30%, like the Pose controlnet. The input image (canvas) overrules the Pose controlnet, while i.e. the Canny Edge controlnet is not overruled.

Edit, it's easily overlooked, but buttons can get extra options. For example, the generate button:
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/eb68bad8-f4ce-44b4-b0fb-ab8bb6577a51)

... becomes a Fill button when a selection is made.
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/c73d9936-0a03-443f-a533-da92add2de83)

And when you click on its drop down button on the right you get more options
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/df8ac432-02fa-4aef-b76a-6e000ef8daf9)

... where some options have options
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/e357b893-4f73-459f-8076-21f4fa81ce93)

... and even more options
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/b3a1b72c-4353-4ad3-bc79-fd7bf20093dd)
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/c28f79ee-0947-4163-bd0f-d038d68d9e36)

I played save by not relying on text prompt for inpainting too much, but I missed the options-options-options. With default settings I couldn't add a mouse in a selection with the text prompt, but by setting the proper context it worked.
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/3fad7878-4275-4a34-82c3-fd55d109613d)

PS, I used 'Generate (Custom)' to add the mouse. The mode 'Add Content' doesn't always work and can give weird results.

**Comment by ETCore7** [comment]
(2) Here is a depth map operation.

Original image, an AI generated cat and an inpainted mouse in the back:

![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/43b681fa-dfce-49ef-aff2-3721f9ffbfc1)

> Add Depth controlnet and click 
![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/d044f885-3256-4ee9-9cde-ffbc47b28b46)
 to generate a controlnet layer.

![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/fe874046-adc6-4679-bbcd-780026406ecc)

In Live mode with text prompt empty, make the dark blob lighter and a mouse (sort of) appears more to the front.

![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/55e5f750-e3c9-442f-a1dd-1109da5e4fe5)

In case the original image should not be altered anywhere else you can lasso the blob and copy the result to the canvas as a new layer.

![image](https://github.com/Acly/krita-ai-diffusion/assets/150020319/b646e66c-ba48-44f9-8c57-e84156899d19)
